### PR TITLE
160577578 Pagination of articles with infinite scrolling

### PR DESCRIPTION
**Pull request by gitaumoses4**
**What does this PR do?**
Implements the pagination logic for articles list display

**Description of Task to be completed?**

Have the homepage contain infinite scrolling for pagination
Have a list of published 

**How should this be manually tested?**

Clone the repository to your local environment.
```
git clone https://github.com/andela/ah-codeblooded-frontend.git
```

Change directories
```
cd ah-codeblooded-frontend/
```

Install the dependencies
```
npm install
```

Run the server
```
yarn start
```

It will open a browser with the current route at `/` and view the list of articles.

**Any background context you want to provide?**

None

**What are the relevant pivotal tracker stories?**

[#160577578](https://www.pivotaltracker.com/story/show/160577578)

**Screenshots (if appropriate)**

###A rtcle Infinite scroll on home page for pagination
screen shot 2018-11-14 at 12 05 42

### List of drafts on the page /me/articles/
screen shot 2018-11-14 at 12 05 50

### List of published articles on the page /me/articles
screen shot 2018-11-14 at 12 05 53

**Questions:**

None

### CODE REVIEW

File: src/components/MyArticles/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  filter = (articles, published) => {
+    const result = articles.filter(article => article.published === published);
+    if (published) {
+      this.setState({ published: result.length, drafts: articles.length - result.length });
+    }
+    return result;
+  };
+
+  filterDrafts = articles => this.filter(articles, false);
```
Comment (paulrimiru) [inline src/components/MyArticles/index.js:29]: Write tests to cover this line

File: src/components/MyArticles/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  filterPublished = articles => this.filter(articles, true);
```
Comment (paulrimiru) [inline src/components/MyArticles/index.js:31]: Add tests to cover this line

File: src/containers/ArticleListing/index.js [hunk 3]
```diff
@@ -23,41 +34,92 @@ class ArticleListing extends React.Component {
...
+    const { page } = this.state;
+    return (
+      !articles || articles.length === 0 ? (
+        this.renderNoItems()
+      )
+        : (
+          <InfiniteScroll
+            pageStart={pageStart}
+            initialLoad={false}
+            loadMore={p => this.loadMore(p, totalPages)}
```
Comment (paulrimiru) [inline src/containers/ArticleListing/index.js:63]: Add tests to cover this line

**Review by paulrimiru — CHANGES_REQUESTED** [review]
Good work, the lines I have indicated require special attention as they are part of the core feature, add tests to cover these lines

File: src/containers/ArticleListing/index.js (hunk 3 above)
Comment (gitaumoses4) [reply src/containers/ArticleListing/index.js:63]: Thanks for the review. Let me work on it.

Comment (gitaumoses4) [reply src/containers/ArticleListing/index.js:63]: I have implemented the changes as requested.

**Review by paulrimiru — APPROVED** [review]

**Pull request merged**